src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -370,19 +370,11 @@ private static SFLoginOutput newSession(
       throw new SFException(ex, ErrorCode.INTERNAL_ERROR, "unexpected URI syntax exception:1");
     }
 
-    if (loginInput.getServerUrl().indexOf(".privatelink.snowflakecomputing.com") > 0) {
-      // Privatelink uses special OCSP Cache server
-      try {
-        URL url = new URL(loginInput.getServerUrl());
-        String host = url.getHost();
-        logger.debug("HOST: {}", host);
-        String ocspCacheServerUrl =
-            String.format("http://ocsp.%s/%s", host, SFTrustManager.CACHE_FILE_NAME);
-        logger.debug("OCSP Cache Server for Privatelink: {}", ocspCacheServerUrl);
-        resetOCSPResponseCacherServerURL(ocspCacheServerUrl);
-      } catch (IOException ex) {
-        throw new SFException(ex, ErrorCode.INTERNAL_ERROR, "unexpected URL syntax exception");
-      }
+    try {
+      // Adjust OCSP cache server if it is private link
+      resetOCSPUrlIfNecessary(loginInput.getServerUrl());
+    } catch (IOException ex) {
+      throw new SFException(ex, ErrorCode.INTERNAL_ERROR, "unexpected URL syntax exception");
```

Comment:
nit; INTERNAL_ERROR sounds weird as the input URL comes from the application. If a wrong URL is given, it should raise an actionable error.

Reply:
Yes, it is some kind of weird. I will change it as ErrorCode.IO_ERROR since the exception is IOException.
In practice, I think this issue should not happen, the URI is validated before calling resetOCSPUrlIfNecessary().